plugins/modules/zos_mount.py

```diff
@@ -1040,45 +1015,25 @@ def run_module(module, arg_def):
                 stderr=str(res_args),
             )
 
-        tmp_file = tempfile.NamedTemporaryFile(delete=True)
-        tmp_file_filename = tmp_file.name
-        tmp_file.close()
+        bk_ds = datasets.tmp_name(high_level_qualifier=tmphlq)
+        datasets.create(name=bk_ds, dataset_type="SEQ")
 
-        copy_uss_mvs(data_store, tmp_file_filename, is_binary=False)
+        new_str = get_str_to_keep(dataset=data_store, src=src)
 
-        module.run_command(
-            "chtag -tc ISO8859-1 " + tmp_file_filename, use_unsafe_shell=False, errors='replace'
-        )
+        for line in new_str:
+            datasets.write(dataset_name=bk_ds, content=line.rstrip(), append=True)
 
-        with open(tmp_file_filename, "r") as fh:
-            content = fh.read().splitlines()
+        datasets.delete(dataset=data_store)
+        datasets.copy(source=bk_ds, target=data_store)
+        datasets.delete(dataset=bk_ds)
```

Comment:
If something goes wrong while we are writing into the backup we should always delete the backup